### ISSUE TITLE
Cannot remove installed package sometimes (MSYS shell)

### ISSUE THREAD

**Issue by annulen** [conan-io/conan#653]
Sometimes when trying to remove package in MSYS shell I get the following error
```
$ conan remove libxml2/2.9.4@annulen/testing
Are you sure you want to delete from 'libxml2/2.9.4@annulen/testing' (yes/no): yes
Traceback (most recent call last):
  File "<string>", line 10, in <module>
  File "<string>", line 6, in run
  File "conan\conans\client\command.py", line 861, in main
  File "conan\conans\client\command.py", line 772, in run
  File "conan\conans\client\output.py", line 86, in error
  File "conan\conans\client\output.py", line 53, in writeln
  File "conan\conans\client\output.py", line 67, in write
UnicodeEncodeError: 'ascii' codec can't encode characters in position 123-129: ordinal not in range(128)
conan returned -1
```

**Comment by annulen** [comment]
```
$ conan --version
Conan version 0.15.0
```

**Comment by memsharded** [comment]
It seems that some error message (most likely some German character) is failing to be encoded. ``conan remove`` error messages are typical in Windows, when one file or folder is busy or opened.

With that trace, I have not enough info, I will try to hunt for the bug, but if you could try to run with ``set CONAN_LOGGING_LEVEL=10`` that might help too for these cases.

**Comment by memsharded** [comment]
As a temporary workaround, these ``conan remove`` errors are typically solved just closing files and repeating the command, or manually removing from ``./conan/data`` folder.

**Comment by memsharded** [comment]
Which conan are you running? Not the version, the installer. With which python version? Thanks!

**Comment by annulen** [comment]
Official 0.15.0 windows build

**Comment by annulen** [comment]
It's quite possible that it tried to remove open directory or file and just failed to print localized error message to console. I just haven't guessed it from error I got :)

**Comment by memsharded** [comment]
So python 2.7, then, if the official windows installer. And yes, the problem is that ``remove`` is launching a ConanException informing about the open file, that wraps a OS message with the offending file/folder, but the message itself is failing to print because of the encoding :)

I am running some tests, trying to fake such a message and force the same behavior, still no luck, but, I hope I am able to find it.

**Comment by annulen** [comment]
Log with CONAN_LOGGING_LEVEL=10:
![conan](https://cloud.githubusercontent.com/assets/102544/20308160/14fab60e-ab5c-11e6-878c-999e41cd1548.png)

**Comment by annulen** [comment]
Reproduced with cmd.exe shell
Log with CONAN_LOGGING_LEVEL=10:
![conan](https://cloud.githubusercontent.com/assets/102544/20308160/14fab60e-ab5c-11e6-878c-999e41cd1548.png)
Sorry cannot copy text properly from cmd

**Comment by memsharded** [comment]
What a nightmare!

All the possible combinations of Py2/3, output terminal encoding (cp850, None for file output, utf8 in development IDEs...), encodings from OS messages.... Then the StringIO used for testing and capturing output has a different interface, only accepting unicode.... After many hours I have submitted a PR #671 that might help. 

I have added some test, but real testing is almost impossible, so if you want to run from sources and check it, the branch is: https://github.com/memsharded/conan/tree/feature/fix_encodings

**Comment by memsharded** [comment]
This should be fixed with the PR, will be released in next 0.16

**Comment by memsharded** [comment]
Released in 0.16
